### ISSUE TITLE
Improve batchNorm

### ISSUE THREAD

**Issue by zolkis** [webmachinelearning/webnn#334]
For discussing how to improve the [BatchNorm section](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-batchnorm). I am working on a draft PR.

There is a good note/example, and description of the parameters.

Formal steps are missing. They should spec how the input and output is handled. For instance, whether normalization can be done in-situ. May the impl modify the input and return the same object (I assume not, i.e. different JS objects, but may share underlying buffers).

In addition, I thought to add the batch norm algorithm from the linked paper as an informal, descriptive sub-steps section.

Also, somewhat more detailed introduction, including typical use cases (currently shortly mentioned).

Any inputs, examples, discussion welcome.

**Comment by anssiko** [comment]
Note: we are establishing a blueprint for formal op definition conventions here using `batchNorm` as a guinea pig. We will scale this effort to the entire set of ops, this is tracked in https://github.com/webmachinelearning/webnn/issues/210#issuecomment-1326361748

Feedback at this stage is important to ensure timely progress.

**Comment by zolkis** [comment]
I am preparing a PR which
- adds more text in the description, in particular clarify how "axis" is used (it is used differently in various frameworks)
- validate the inputs (dimensions etc)
- add an algorithm (which "outsources" running the actual op, but informatively describes what's expected).
It will remain WiP until some other algorithms will be merged.

**Comment by inexorabletash** [comment]
This looks completed by the merged PRs, so closing. If there is work remaining, please re-open or file a new issue.
